core/src/json_schema.rs

```diff
@@ -282,6 +363,297 @@ pub mod simplified {
 
 #[cfg(test)]
 mod unit {
+    macro_rules! remove_multiple_test {
+        ($func_name:ident, expected = $expected:expr, input = $input:expr) => {
+            #[test]
+            fn $func_name() {
+                let res = $input.remove_multiple().unwrap();
+
+                assert_eq!(res, $expected);
+            }
+        };
+    }
+
+    // Testing TryFrom<json_schema::Schema> for json_schema::simplified::Atomic
+    mod remove_multiple {
```

Comment:
Other test ideas if you think they are worthwhile:
- More than 2 types in multiple
- Array with Items: Some(Multiple[...]) that turns into a Some(Single(Any))